### PR TITLE
Ref135634 call api list using device, list using device by status and search

### PR DESCRIPTION
**Pull request by lamvuvan**

### CODE REVIEW

File: app/src/main/java/com/framgia/fdms/data/source/DeviceUsingHistoryRepository.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import java.util.List;
+
+import rx.Observable;
+
+/**
+ * Created by lamvu on 11/09/2017.
+ */
+
+public class DeviceUsingHistoryRepository implements DeviceUsingHistoryDataSource {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/fdms/data/source/DeviceUsingHistoryRepository.java]: WARNING: Class DeviceUsingHistoryRepository should be declared as final.

File: app/src/main/java/com/framgia/fdms/data/source/remote/DeviceUsingHistoryRemoteDataSource.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        mApi = FDMSServiceClient.getInstance();
+    }
+
+    @Override
+    public Observable<List<DeviceUsingHistory>> getListDeviceUsingHistory(String status,
+                                                                          String name,
+                                                                          String email,
+                                                                          String code) {
+        return mApi.getDeviceUsingHistories(status, name, email, code)
+                .flatMap(new Func1<Respone<List<DeviceUsingHistory>>, Observable<List<DeviceUsingHistory>>>() {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/fdms/data/source/remote/DeviceUsingHistoryRemoteDataSource.java]: ERROR: Line is longer than 100 characters (found 111).

File: app/src/main/java/com/framgia/fdms/data/source/remote/DeviceUsingHistoryRemoteDataSource.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                    @Override
+                    public Observable<List<DeviceUsingHistory>> call(Respone<List<DeviceUsingHistory>> listRespone) {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/fdms/data/source/remote/DeviceUsingHistoryRemoteDataSource.java]: ERROR: Line is longer than 100 characters (found 117).

File: app/src/main/java/com/framgia/fdms/data/source/remote/DeviceUsingHistoryRemoteDataSource.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    public Observable<List<DeviceUsingHistory>> getListDeviceUsingHistory(String status,
+                                                                          String name,
+                                                                          String email,
+                                                                          String code) {
+        return mApi.getDeviceUsingHistories(status, name, email, code)
+                .flatMap(new Func1<Respone<List<DeviceUsingHistory>>,
+                        Observable<List<DeviceUsingHistory>>>() {
+                    @Override
+                    public Observable<List<DeviceUsingHistory>>
+                    call(Respone<List<DeviceUsingHistory>> listRespone) {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/fdms/data/source/remote/DeviceUsingHistoryRemoteDataSource.java]: ERROR: 'call' have incorrect indentation level 20, expected level should be 24.

**Comment by lamvuvan** [comment]
new pull

**Pull request closed without merging**